backend/app/adapter/gqlapi/resolver/authmutation.go

```diff
@@ -17,16 +17,35 @@ type AuthMutation struct {
 	authenticator authenticator.Authenticator
 	changeLog     changelog.ChangeLog
 	urlCreator    url.Creator
+	urlUpdater    url.Updater
 }
 
 // URLInput represents possible ShortLink attributes
 type URLInput struct {
-	OriginalURL string
+	OriginalURL *string
 	CustomAlias *string
 	ExpireAt    *time.Time
 }
 
-// CreateURLArgs represents the possible parameters for CreateShortLink endpoint
+func (u *URLInput) isEmpty() bool {
```

Comment:
I definitely agree that the use case should handle this business logic instead of having the resolver address it directly. I'll attempt to refactor the URL package so that the boundaries of responsibility are respected.